_drafts/release-workflows.md

```diff
@@ -0,0 +1,388 @@
+
+Note: By default lts releases are published on the default channel as they are limited to a range of versions inferior to the versions released from `master`. Dependents will get only the expected releases by specifying a range dependency (for example `^2.0.0`).
+If a `channel` is specified, then the release will be made on this channel.
+
+##### LTS branches configuration
+
+```json
+"release": {
+  "branches": ["1.x.x", "2.x.x", "master"]
+  }
+}
+```
+
+With specific channel names:
+
+```json
+"release": {
+  "branches": [{
+    "branch": "v1",
+    "range": "1.x.x",
+    "channel": "v1"
+  }, {
+    "branch": "v2",
+    "range": "2.x.x",
+    "channel": "v2"
+  }, "master"]
+}
+```
+
+#### Pre-releases
+
+Workflow to distribute releases without incrementing the semantic version during the development of new a application version.  
+Ideal for applications distributing unstable/alpha/preview releases.
+
+Branch: `master`, `next-beta`, `next-alpha`
+
+Channel: `latest` if supported by the release target
+
+##### Pre-releases workflow
+
+| Action                                                         | Result                                                                                                                                        |
+|----------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------|
+| **push** to `master`                                           | Release on default channel (if supported, otherwise no channel)                                                                               |
+| **merge** `4.0.0-beta`, `5.0.0-beta` or any branch to `master` | Release on default channel (if supported, otherwise no channel)                                                                               |
+| **push** to `4.0.0-beta`                                       | Release a prerelease version (`4.0.0-beta`, then `4.0.0-beta.1`, then `4.0.0-beta.2`) on default channel (if supported, otherwise no channel) |
+| **merge** any branch to `4.0.0-beta`                           | Release a prerelease version (`4.0.0-beta`, then `4.0.0-beta.1`, then `4.0.0-beta.2`) on default channel (if supported, otherwise no channel) |
+| **push** to `5.0.0-beta`                                       | Release a prerelease version (`5.0.0-beta`, then `5.0.0-beta.1`, then `5.0.0-beta.2`) on default channel (if supported, otherwise no channel) |
+| **merge** any branch to `5.0.0-beta`                           | Release a prerelease version (`5.0.0-beta`, then `5.0.0-beta.1`, then `5.0.0-beta.2`) on default channel (if supported, otherwise no channel) |
+
+##### Pre-releases configuration
+
+```json
+"release": {
+  "branches": ["master", "4.0.0-beta", "5.0.0-alpha"]
```

Comment:
I wonder if we need the branches configuration? What if I forget about changing the `branches` setting and push a branch `4.1.0-beta`?

Reply:
Then the branch will be ignored, on considered as a "non-release branch".

We could auto-determine everything by branch name. That's way more powerful.
But automating to much can create unwanted behavior.

As we are dealing with the sensitive subject of releases, I prefer to do stuff only when the user explicitly wants to and configure the branch. But that's really up to debate.

Reply:
i can see both sides. i wouldnt want all of my branches to automatically publish versions for every commit, but would instead want to opt in. however, i could see there being multiple ways to opt in. 

* through config would be the most explicit and would require me to follow any conventions but my own in naming the branches
* following branch naming conventions could be convenient so that a strict subset of branch names would automatically get the desired behavior. there might be cases where a branch gets named in a way that matches one of the semantic-release conventions unintenionally and a version gets published unexpectedly. since it wouldn't be tagged as `latest`, though, its probably low impact. i also wonder if some projects would use some of these conventions from branch names w/out the intention for publishing. 

maybe there could be top-level config to enable/disable following the conventions automatically so that the opt-in could be per project rather than per-branch? this could then be extended by a [sharable config](https://semantic-release.gitbooks.io/semantic-release/content/docs/usage/shareable-configurations.html) to opt-in at a more org-level.

Reply:
> a version gets published unexpectedly

I agree there is some risk to that. It is similar to the risk we already have with semantic-release using commit conventions to release the right version. A tiny typo in `BREAKCING CHANGE:` and you accidentally release a breaking change in a fix version. 

The way I would address it is to limit the conventions we support to only `1.x` (etc), `latest`, `next` and document them very clearly.

There is always a risk between convenience and unintended behavior. I think semantic-release is striking that balance very well today and it certainly had a big impact in terms of promoting commit message conventions. I think we can achieve the same with branch name conventions when we support some of them out of the box. It carves the way for best practises and helps to see recurring patterns across projects which is great for Open Source overall